### PR TITLE
IDEX-3913: Implement stack api

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
@sleshchenko @gazarenkov  @benoitf  @ashumilova @evoevodin @skabashnyuk

### CODE REVIEW

File: assembly-ide-war/src/main/resources/predefined-stacks.json [hunk 1]
```diff
@@ -0,0 +1,711 @@
...
+      "description":null,
+      "commands":[
+        {
+          "commandLine":"mvn clean install -f ${current.project.path}",
+          "name":"newMaven",
+          "type":"mvn"
+        }
+      ]
+    },
+    "iconLink":"http://path.com/to-the-icon.svg",
```
Comment (benoitf) [inline assembly-ide-war/src/main/resources/predefined-stacks.json]: ? as image is embedded we don't need this 'example link'

File: assembly-ide-war/src/main/resources/predefined-stacks.json [hunk 2]
```diff
@@ -0,0 +1,711 @@
...
+            "read",
+            "search"
+          ],
+          "name": "public"
+        }
+      ],
+      "users": {}
+    }
+  }
+]
```
Comment (benoitf) [inline assembly-ide-war/src/main/resources/predefined-stacks.json]: missing newline at the end of file

File: assembly-ide-war/src/main/resources/predefined-stacks.json (hunk 1 above)
Comment (AndrienkoAleksandr) [inline assembly-ide-war/src/main/resources/predefined-stacks.json]: Ok, I'll fix it.

File: assembly-ide-war/src/main/resources/predefined-stacks.json (hunk 2 above)
Comment (AndrienkoAleksandr) [inline assembly-ide-war/src/main/resources/predefined-stacks.json]: Ok, I'' fix it.

**Comment by garagatyi** [comment]
Please group all bindings you added because they all are needed for the same feature

**Comment by riuvshin** [comment]
Not OK, predefined-stacks.json is duplicated for all the packagings che, cloud-ide, onpremises which is not easy to support. My proposal is to move it to che-core as we did for templates https://github.com/codenvy/che-core/tree/master/ide/che-core-ide-templates then use it on packaging stage

**Pull request closed without merging**